tensorboard/webapp/plugins/npmi/views/annotations_list/legend/legend_element/legend_element_component.ts

```diff
@@ -0,0 +1,73 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  AfterViewInit,
+  ViewChild,
+  ElementRef,
+} from '@angular/core';
+import * as d3 from 'd3';
+
+@Component({
+  selector: 'npmi-legend-element',
+  templateUrl: './legend_element_component.ng.html',
+  styleUrls: ['./legend_element_component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class LegendElementComponent implements AfterViewInit {
+  @Input() text!: string;
+  @Input() color!: string;
+  @Input() shape!: string;
+  @ViewChild('glyph', {static: true, read: ElementRef})
+  private readonly glyphSVG!: ElementRef<SVGElement>;
+  // Drawing containers
+  private svg: any;
+  private mainContainer: any;
+
+  ngAfterViewInit(): void {
+    this.svg = d3.select(this.glyphSVG.nativeElement);
+    this.mainContainer = this.svg.append('g');
+    this.draw();
+  }
+
+  private draw() {
```

Comment:
I find this current implementation for choosing and drawing shapes a little smelly but I can't quite put my finger on what the source of the smell is. It's fine to do like this but I guess I'm just reserving the right to revisit later.